vars/containerExecuteStructureTests.groovy

```diff
@@ -0,0 +1,134 @@
+import com.sap.piper.Utils
+import groovy.transform.Field
+
+@Field String STEP_NAME = getClass().getName()
+
+@Field Set GENERAL_CONFIG_KEYS = [
+    /**
+     * Print more detailed information into the log.
+     * @possibleValues `true`, `false`
+     */
+    'verbose'
+
+]
+
+@Field Set STEP_CONFIG_KEYS = GENERAL_CONFIG_KEYS.plus([
+    /**
+     * Only for Kubernetes environments: Command which is executed to keep container alive, defaults to '/usr/bin/tail -f /dev/null'
+     */
+    'containerCommand',
+    /**
+     * Only for Kubernetes environments: Shell to be used inside container, defaults to '/bin/sh'
+     */
+    'containerShell',
+    /**
+      * Docker image for code execution.
+      */
+    'dockerImage',
+    /**
+     * Options to be passed to Docker image when starting it (only relevant for non-Kubernetes case).
+     */
+    'dockerOptions',
+    /**
+      * Defines the behavior, in case tests fail.
+      * @possibleValues `true`, `false`
+      */
+    'failOnError',
+    /**
+     * Only relevant for testDriver 'docker'.
+     * @possibleValues `true`, `false`
+     */
+    'pullImage',
+    /**
+      * If specific stashes should be considered for the tests, you can pass this via this parameter.
+      */
+    'stashContent',
+    /**
+     * Container structure test configuration in yml or json format. You can pass a pattern in order to execute multiple tests.
+     */
+    'testConfiguration',
+    /**
+     * Container structure test driver to be used for testing, please see https://github.com/GoogleContainerTools/container-structure-test for details.
+     */
+    'testDriver',
+    /**
+    * Image to be tested
+    */
+    'testImage',
+    /**
+     * Path and name of the test report which will be generated
+     */
+    'testReportFilePath',
+])
+
+@Field Set PARAMETER_KEYS = STEP_CONFIG_KEYS
+
+void call(Map parameters = [:]) {
+    handlePipelineStepErrors(stepName: STEP_NAME, stepParameters: parameters) {
+
+        def script = checkScript(this, parameters) ?: this
+
+        def utils = parameters?.juStabUtils ?: new Utils()
+
+        // load default & individual configuration
+        Map config = ConfigurationHelper.newInstance(this)
+            .loadStepDefaults()
+            .mixinGeneralConfig(script.commonPipelineEnvironment, GENERAL_CONFIG_KEYS)
+            .mixinStepConfig(script.commonPipelineEnvironment, STEP_CONFIG_KEYS)
+            .mixinStageConfig(script.commonPipelineEnvironment, parameters.stageName?:env.STAGE_NAME, STEP_CONFIG_KEYS)
+            .mixin(parameters, PARAMETER_KEYS)
+            .addIfEmpty('testDriver', Boolean.valueOf(script.env.ON_K8S) ? 'tar' : 'docker')
+            .addIfNull('pullImage', !Boolean.valueOf(script.env.ON_K8S))
+            .withMandatoryProperty('dockerImage')
+            .use()
+
+        utils.pushToSWA([step: STEP_NAME], config)
+
+        if (config.verbose) echo "[${STEP_NAME}] Configuration: ${config}"
```

Comment:
would be cool to add this to the ConfigurationHelper itself i.e. upon the invocation of `use()`

Reply:
Fully agree, There will be a separate PR to address this ...

Reply:
Yep, just got https://github.com/SAP/jenkins-library/pull/459 for review;-)